### ISSUE TITLE
Feature request: Tooltip enable check

### ISSUE THREAD

**Issue by Quinn-L** [ng-bootstrap/ng-bootstrap#737]
LIke ui-bootstrap, having an `enable` function would help in the conditionally showing the tooltip.

My personal use case is checking if there is anything to display, otherwise I end up seeing just a tooltip arrow.

**Comment by jnizet** [comment]
I can do that, but we need to decide on two points:
- how to name the input. `enabled` looks too broad to me, especially for an element that could have other directives. `ngbEnabled`could clash with one of our other directives that would need the same kind of input. `ngbTooltipEnabled`? `tooltipEnabled`?
- should we close the tooltip if if it's already opened when the input becomes false? I guess yes.

**Comment by pkozlowski-opensource** [comment]
I'm kind of not sold on the idea of having this new property as it makes API surface larger for a use-cases that I'm not fully sure I understand. @Quinn-L are you binding dynamic value to the toolitp content - a value that can come from a user? Could you share more info on your use-case?

In any case it is possible to have full control over tooltip opening / closing with manual triggers so I'm not sure if the new option "pays for itself". Let's hear more on the @Quinn-L use-cases before we decide to move forward.

**Comment by Quinn-L** [comment]
Yes, I'm using (probably abusing) the tooltip in a dynamic (and editable) datatable, and just showing the cell value as a tooltip (eg. because the cell can't always show the entire value).

It's a bit clunky but I'll investigate binding the trigger.

**Comment by pkozlowski-opensource** [comment]
@Quinn-L what you could do today is this: `<div ngbTooltip={{contentExp}} triggers="manual" #t="ngbTooltip" (mouseenter)="contentExp && t.open()" (mouseleave)="t.close()">`. Obviously you can jave more complex condition in the `(mouseenter)` event handler. 

I understand that the above is a bit wordy but then again, I don't think this is a "typical" use-case.

What we can (and probably should!) to do instead of the `enabled` input it is to check the argument passed to the `ngbTooltip` and don't open a tooltip if it is `null` / `undefined` / empty string. Would this work for you?

**Comment by Quinn-L** [comment]
Thanks for the workaround.

Yep, I think not showing the tooltip if supplied null/undefined/'' would make a shorter workaround for disabling the tooltip in my case.

**Comment by pkozlowski-opensource** [comment]
> Yep, I think not showing the tooltip if supplied null/undefined/'' would make a shorter workaround for disabling the tooltip in my case.

@jnizet do you want to take a stab at this one?

**Comment by jnizet** [comment]
Yes, I can. Just to be clear, if the user wants to conditionally pass an html template, he would have to use something like the following, right?

```
<template #tipContent>Hello, <b>{{name}}</b>!</template>
<button type="button" class="btn btn-secondary" [ngbTooltip]="shouldDisplayTooltip ? tipContent : null">
  I've got markup and bindings in my tooltip!
</button>
```

or something like the following:

```
<template *ngIf="shouldDisplayTooltip" #tipContent>Hello, <b>{{name}}</b>!</template>
<button type="button" class="btn btn-secondary" [ngbTooltip]="tipContent">
  I've got markup and bindings in my tooltip!
</button>
```

Should we close the tooltip if it's already opened and its content becomes null?

**Comment by pkozlowski-opensource** [comment]
> Should we close the tooltip if it's already opened and its content becomes null?

I would say yes.

**Comment by anrepppp** [comment]
I am working on displaying popover when there is content. I have came across this case where there is no content and a tooltip arrow is showing. I have followed [ngbPopover]="list.length > 0 ? pdqContent : null" way and other approach mentioned by jnizet. It is still showing the arrow.

**Comment by pkozlowski-opensource** [comment]
@anrepppp commenting on closed issues is pointless. Either ask a support question on an [appropriate forum](https://github.com/ng-bootstrap/ng-bootstrap#getting-help) or just open a new issue [following the issue template[(https://github.com/ng-bootstrap/ng-bootstrap#you-think-youve-found-a-bug)

**Comment by anrepppp** [comment]
I am sorry, open a new ticket for this issue.
